tt_eager/tt_dnn/op_library/composite/composite_ops.cpp

```diff
@@ -697,12 +701,27 @@ Tensor subalpha(const Tensor& input_a, const Tensor& input_b, float alpha, const
 }
 
 // addalpha(input, other, alpha) = input + (alpha * other)
-Tensor _addalpha(const Tensor& input_a, const Tensor& input_b, float alpha, const MemoryConfig& output_mem_config) {
-    Tensor result = add(mul_unary(input_b, alpha, output_mem_config), input_a, std::nullopt, output_mem_config);
-    return result;
+Tensor _addalpha(
+    const Tensor& input_a,
+    const Tensor& input_b,
+    float alpha,
+    std::optional<Tensor> output_tensor,
+    const MemoryConfig& output_mem_config) {
+    if (output_tensor.has_value()) {
+        add(mul_unary(input_b, alpha, output_mem_config), input_a, std::nullopt, output_tensor.value().memory_config(), std::nullopt, output_tensor.value());
```

Comment:
Ideally the output_mem_config arg should be made an optional for binary as well so that we can just pass std::nullopt, but since it isn't you can pass `operation::DEFAULT_OUTPUT_MEMORY_CONFIG` instead of querying the output tensor's mem config since querying it is more expensive and the value is not needed

Reply:
We should file an issue to make it optional if we don't have an open issue already

Reply:
I will be doing it as part of this issue https://github.com/tenstorrent/tt-metal/issues/8962